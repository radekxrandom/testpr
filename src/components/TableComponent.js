import React from "react";

/**
 * Component used to display table with company information.
 * @component
 * @param {Object[]} displayedCompanies - Companies to display
 * @param {string} displayedCompanies[].name - Company name
 * @param {string} displayedCompanies[].city - Company city
 * @param {number} displayedCompanies[].avg - Company average income
 * @param {number} displayedCompanies[].total - Company total income
 * @param {number} displayedCompanies[].latest - Company last month's income
 * @param onClickSortByField - Function which sorts data by clicked field.
 */

const TableComponent = props => {
  return (
    <table>
      <thead>
        <tr id="col">
          <th onClick={() => props.onClickSortByField("id")}>
            <p className="controls">
              <span className="tblHdTxt">ID </span>
              {props.arrow === "id" && <span className="arrow"> &darr;</span>}
              {props.arrow === "idReversed" && (
                <span className="arrow"> &uarr;</span>
              )}
            </p>
          </th>
          <th onClick={() => props.onClickSortByField("name")}>
            <p className="controls">
              <span className="tblHdTxt">NAME </span>
              {props.arrow === "name" && <span className="arrow"> &darr;</span>}
              {props.arrow === "nameReversed" && (
                <span className="arrow"> &uarr;</span>
              )}
            </p>
          </th>
          <th onClick={() => props.onClickSortByField("city")}>
            <p className="controls">
              <span className="tblHdTxt">CITY </span>
              {props.arrow === "city" && <span className="arrow"> &darr;</span>}
              {props.arrow === "cityReversed" && (
                <span className="arrow"> &uarr;</span>
              )}
            </p>
          </th>
          <th onClick={() => props.onClickSortByField("total")}>
            <p className="controls">
              <span className="tblHdTxt">TOTAL </span>
              {props.arrow === "total" && (
                <span className="arrow"> &darr;</span>
              )}
              {props.arrow === "totalReversed" && (
                <span className="arrow"> &uarr;</span>
              )}
            </p>
          </th>
          <th onClick={() => props.onClickSortByField("avg")}>
            <p className="controls">
              <span className="tblHdTxt">AVERAGE </span>
              {props.arrow === "avg" && <span className="arrow"> &darr;</span>}
              {props.arrow === "avgReversed" && (
                <span className="arrow"> &uarr;</span>
              )}
            </p>
          </th>
          <th onClick={() => props.onClickSortByField("latest")}>
            <p className="controls">
              <span className="tblHdTxt">LATEST </span>
              {props.arrow === "latest" && (
                <span className="arrow"> &darr;</span>
              )}
              {props.arrow === "latestReversed" && (
                <span className="arrow"> &uarr;</span>
              )}
            </p>
          </th>
        </tr>
      </thead>
      <tbody>
        {props.displayedCompanies.length > 0 &&
          props.displayedCompanies.map(row => (
            <tr key={row.id}>
              <td>{row.id}</td>
              <td>{row.name}</td>
              <td>{row.city}</td>
              <td>{row.total}</td>
              <td>{row.avg}</td>
              <td>{row.latest}</td>
            </tr>
          ))}
        {!props.displayedCompanies.length && (
          <tr>
            <td align="center" className="tdSpanWholeRow" colspan="6">
              No search results
            </td>
          </tr>
        )}
      </tbody>
    </table>
  );
};

export default TableComponent;
