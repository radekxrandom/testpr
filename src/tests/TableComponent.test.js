import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import TableComponent from "../components/TableComponent";

Enzyme.configure({ adapter: new Adapter() });

const mockData = [
  {
    avg: 5200.28,
    city: "North Shirleyfurt",
    id: 1,
    latest: 2652,
    name: "Ernser Group",
    total: 260014
  },
  {
    id: 2,
    name: "Franecki, Torphy and Lesch",
    city: "Port Halle",
    total: 251365,
    avg: 5027.3,
    latest: 7060
  },
  {
    id: 2,
    name: "Franecki, Torphy and Lesch",
    city: "Port Halle",
    total: 251365,
    avg: 5027.3,
    latest: 7060
  },
  {
    id: 4,
    name: "Smitham Group",
    city: "Rennerfort",
    total: 255396,
    avg: 5107.92,
    latest: 3167
  },
  {
    id: 6,
    name: "Towne - Vandervort",
    city: "Cassidyfurt",
    total: 283957,
    avg: 5679.14,
    latest: 1240
  }
];

describe("TblComponent", () => {
  test("renders ok and works ok", () => {
    const mockClick = jest.fn();
    const output = shallow(
      <TableComponent rowObjects={mockData} arrow="id" sortBy={mockClick} />
    );
    output
      .find("th")
      .at(0)
      .simulate("click");
    expect(mockClick).toHaveBeenCalled();
    expect(
      shallow(
        <TableComponent rowObjects={mockData} arrow="id" sortBy={mockClick} />
      )
    ).toMatchSnapshot();
  });
});
