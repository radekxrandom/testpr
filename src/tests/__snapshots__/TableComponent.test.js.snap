// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TblComponent renders ok and works ok 1`] = `ShallowWrapper {}`;
